kogito-editors-js/packages/boxed-expression-component/src/components/Resizer/common/index.ts

```diff
@@ -27,7 +27,20 @@ export const widthValue = (width: number | string | undefined | null): number =>
  * Generates a global supervisor hash for a given object.
  */
 export const hashfy = (obj = {}): string => {
-  return JSON.stringify(obj);
+  const getCircularReplacer = () => {
+    const seen = new WeakSet();
+    return (key: string, value: unknown) => {
+      if (typeof value === "object" && value !== null) {
+        if (seen.has(value)) {
+          return;
+        }
+        seen.add(value);
+      }
+      return value;
+    };
+  };
+
+  return JSON.stringify(obj, getCircularReplacer());
```

Comment:
Since `jsInterop` layer adds circular references inside Java objects that are translated into JS Objects, the `stringify` method would fail with such circular references. There is a proper solution provided by MDN (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cyclic_object_value#examples)